src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,214 @@
+        return unlink(index);
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        int removeByIndex = getIndexIfContains(object);
+        if (removeByIndex == -1) {
+            return false;
+        }
+        unlink(removeByIndex);
+        return true;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void linkLast(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (head == null) {
+            head = tail = newNode;
+        } else {
+            tail.next = newNode;
+            newNode.prev = tail;
+            tail = newNode;
+        }
+        size++;
+    }
+
+    private void linkFirst(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        newNode.next = head;
+        head.prev = newNode;
+        head = newNode;
+        size++;
+    }
+
+    private void link(T value, int index) {
+        Node<T> newNode = new Node<>(null, value, null);
+        Node<T> previous = getNodeByIndex(index - 1);
+        newNode.next = previous.next;
+        previous.next.prev = newNode;
+        previous.next = newNode;
+        newNode.prev = previous;
+        size++;
+    }
+
+    private int getIndexIfContains(T object) {
+        int index = -1;
+        Node<T> current = head;
+        int iteration = 0;
+        while (current != null) {
+            if (current.value == object || current.value != null && current.value.equals(object)) {
+                return iteration;
+            }
+            current = current.next;
+            iteration++;
+        }
+        return index;
+    }
+
+    private T unlink(int index) {
+        if (size == 1) {
+            return unlinkTheOne();
+        }
+        if (index == 0) {
+            return unlinkFirst();
+        }
+        if (index == size - 1) {
+            return unlinkLast();
+        }
+        return unlinkWithin(index);
+    }
+
+    private T unlinkTheOne() {
+        T removedValue;
+        removedValue = head.value;
+        head = tail = null;
+        size--;
+        return removedValue;
+    }
+
+    private T unlinkFirst() {
+        T removedValue;
+        removedValue = head.value;
+        head = head.next;
+        head.prev = null;
+        size--;
+        return removedValue;
+    }
+
+    private T unlinkLast() {
+        T removedValue;
+        removedValue = tail.value;
+        tail.prev.next = null;
+        tail = tail.prev;
+        size--;
+        return removedValue;
+    }
+
+    private T unlinkWithin(int index) {
+        T removedValue;
+        Node<T> previous = getNodeByIndex(index - 1);
+        removedValue = previous.next.value;
+        previous.next = previous.next.next;
+        previous.next.prev = previous;
+        size--;
+        return removedValue;
+    }
```

Comment:
Why are we finding previous node, not with our index?
That's not an 1 linked list.